src/main/kotlin/org/rust/lang/core/completion/RsCommonCompletionProvider.kt

```diff
@@ -39,8 +39,12 @@ object RsCommonCompletionProvider : CompletionProvider<CompletionParameters>() {
         context: ProcessingContext,
         result: CompletionResultSet
     ) {
-        val element = parameters.position.parent as RsReferenceElement
-        if (parameters.position !== element.referenceNameElement) return
+        // Use original position if possible to re-use caches of the real file
+        val position = CompletionUtil.getOriginalElement(parameters.position)
+            ?.takeIf { isAncestorTypesEquals(it, parameters.position) }
```

Comment:
I still think it's enough to check only parents instead of all ancestors but looks like it doesn't affect completion performance so it's ok